### ISSUE TITLE
problem connecting to Microsoft Sql server instance 

### ISSUE THREAD

**Issue by shjb16** [dbeaver/dbeaver#3377]
I'm trying to connect to a Microsoft Sql server database using Dbeaver on macbook but keep getting the following error:
Host: IP address/abc     # abc is the name of my instance
port:1689 
TCP/IP is enabled under SQL SERVER configuration manager
integratedSecurity is set : True under driver properties 

error:

"The TCP/IP connection to the host,port 1689 has failed. Error: "connect timed out. Verify the connection properties, check that an instance of SQL Server is running on the host and accepting TCP/IP connections at the port, and that no firewall is blocking TCP connections to the port."

**Comment by shjb16** [comment]
I tried telnet IP address port  and here is the output:
telnet IP address  1689
Trying IP address...
Connected to IP address.
Escape character is '^]'.

**Comment by nmat** [comment]
on the SQLserver can you check the errorlog to see if the call to the server at all goes through and if there is a error in there that might shed some more light?

**Comment by mrotteveel** [comment]
The Microsoft SQL Server JDBC driver ignores the instance name if a port is specified. See [Building the Connection URL](https://docs.microsoft.com/en-us/sql/connect/jdbc/building-the-connection-url?view=sql-server-2017), specifically:

> For optimal connection performance, you should set the portNumber when you connect to a named instance. This will avoid a round trip to the server to determine the port number. **If both a portNumber and instanceName are used, the portNumber will take precedence and the instanceName will be ignored.** 

 Are you sure that port is the right one?

**Comment by ranko6** [comment]
Maybe a caching issue then.
For test, used port"1434" -> Test connection -> Naturally failed because there is no instance on that port.
Then reverted to the default "1433" -> Test connection -> Should have succeeded but got error about port 1434 which is not the one entered.
It does this even after dbeaver is restarted.

![image](https://user-images.githubusercontent.com/1059136/39311898-705e2182-496e-11e8-8753-140073c6b680.png)

Should I open a new ticket for this since it does not seem to be related to this one then?

**Comment by serge-rider** [comment]
@ranko6 I ca't reproduce that. But yes, if you have some steps to reproduce - please create a separate ticket.

**Comment by jorge30dic** [comment]
I do this and works for me.
![image](https://user-images.githubusercontent.com/8659600/52872042-7b0e7800-3119-11e9-9169-0c5cb7b6b269.png)

**Comment by lflucasferreira** [comment]
Did anyone resolved this issue? I am facing this right now. If there is a solution, could anyone give me the correspondent issue tracker id? Thank you in advance.

**Comment by serge-rider** [comment]
Maybe TCP/IP is not configured properly in your SQL Server instance (JDBC doesn't work without this).
Please check details: https://dbeaver.io/forum/viewtopic.php?f=3&t=624

**Comment by lflucasferreira** [comment]
> Maybe TCP/IP is not configured properly in your SQL Server instance (JDBC doesn't work without this).
> Please check details: https://dbeaver.io/forum/viewtopic.php?f=3&t=624

I figured it out. We need to enable the TCP/IP protocol in SQL Server Configuration manager > SQL Server Network Configuration. So simple!

**Comment by mrotteveel** [comment]
@ironvvill There is no need for that, instead start the SQL Server Browser service. This allows the Microsoft SQL Server JDBC driver to connect to named instances with a random port.

**Comment by bagus-repository** [comment]
For anyone facing this issue with SQLEXPRESS and MSSQLSERVER installed, you can disable or set different TCP/IP port foreach in SQL Server Configuration Manager

**Comment by sssnarl** [comment]
@mrotteveel is correct here all that needs to be done is to enable the SQL Server Browser service and start it.  connected without issue to the named instance after that.

**Comment by darandavis** [comment]
@sssnarl That assumes you have installed SQL Server locally right? What if I just want to connect to a company SQL Server instance and the port defaults to 1433 and ignores my port supplied in the connection settings?

**Comment by pavan-chakka** [comment]
Hello, 

I have  MAC M1 Max with Open JDK 11, and using DBeaver 22.1.2 to connect DB hosted MS SQL Server.   Tried all options listed in above chat, but facing same issue  and getting below error - 

"The TCP/IP connection to the host XXXXXX, port 1433 has failed. Error: "XXXXXX. Verify the connection properties. Make sure that an instance of SQL Server is running on the host and accepting TCP/IP connections at the port. Make sure that TCP connections to the port are not blocked by a firewall.". "

![Screen Shot 2022-08-31 at 5 09 32 PM](https://user-images.githubusercontent.com/85208782/187802123-d87baf6f-0915-4f06-b94b-488bb528bd7e.png)

Below is MS SQL Server Driver settings. Not sure what else to be done, please help if you experienced it and got any solution.

![Screen Shot 2022-08-31 at 5 28 17 PM](https://user-images.githubusercontent.com/85208782/187802640-333ac9e0-8d5f-4edd-b47a-d764be4bfec9.png)


![Screen Shot 2022-08-31 at 5 31 05 PM](https://user-images.githubusercontent.com/85208782/187802653-8e62126d-776d-440c-a099-9a4cf74997cd.png)

**Comment by ivanakhoo** [comment]
@ranko6 Any luck with this? I am facing the same issue now...

**Comment by John-Lenon** [comment]
@ironvvill 
You're right, thanks for your comment. In my case, this was exactly what was happening to me, i was using the wrong port number, I was using the default port 1433, but after checking the current port number of my server instance i realized it was a different number.
